src/mscorlib/shared/System/Numerics/Vector.cs

```diff
@@ -763,6 +765,17 @@ private Vector(ref Register existingRegister)
         {
             this.register = existingRegister;
         }
+
+        /// <summary>
+        /// Constructs a vector from the given span.
+        /// The span must contain at least Vector'T.Count elements.
+        /// </summary>
+        [Intrinsic]
+        public unsafe Vector(Span<T> values)
+            : this()
+        {
```

Comment:
This needs bounds check to verify that there is enough elements in the Span.

Reply:
(Add test for it to your CoreFX PR too.)

Reply:
Working on this. I plan to model the new exception on lines of the ask in issue dotnet/corefx#25344. Whatever is done here will have to be replicated in fix for dotnet/corefx#25344. (Would like to work on that after this.)